src/PointwiseFunctions/AnalyticSolutions/WaveEquation/RegularSphericalWave.cpp

```diff
@@ -0,0 +1,126 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/AnalyticSolutions/WaveEquation/RegularSphericalWave.hpp"
+
+#include <algorithm>
+#include <cmath>
+
+#include "DataStructures/DataBox/Prefixes.hpp"  // IWYU pragma: keep
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "DataStructures/Variables.hpp"           // IWYU pragma: keep
+#include "Evolution/Systems/ScalarWave/Tags.hpp"  // IWYU pragma: keep
+#include "Parallel/PupStlCpp11.hpp"               // IWYU pragma: keep
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/EqualWithinRoundoff.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+namespace ScalarWave {
+namespace Solutions {
+
+template <size_t Dim>
+RegularSphericalWave<Dim>::RegularSphericalWave(
+    std::unique_ptr<MathFunction<1>> profile) noexcept
+    : profile_(std::move(profile)) {}
+
+template <size_t Dim>
+tuples::TaggedTuple<ScalarWave::Pi, ScalarWave::Phi<Dim>, ScalarWave::Psi>
+RegularSphericalWave<Dim>::variables(
+    const tnsr::I<DataVector, Dim>& x, double t,
+    const tmpl::list<ScalarWave::Pi, ScalarWave::Phi<Dim>,
+                     ScalarWave::Psi> /*meta*/) const noexcept {
+  const DataVector r = get(magnitude(x));
+  Scalar<DataVector> psi{r.size()};
+  Scalar<DataVector> dpsi_dt{r.size()};
+  tnsr::i<DataVector, Dim> dpsi_dx{r.size()};
+  // This implementation, as well as the one in the function below, can be
+  // vectorized with the support for [element
+  // selections](https://bitbucket.org/blaze-lib/blaze/wiki/Element%20Selections)
+  // that is added in Blaze v3.3.
```

Comment:
.. or is there anything we can do to vectorize this already with the tools we have?

Reply:
They can't be vectorized on anything other than AVX512 (Xeon Phi), which is already deprecated hardware.